### DISCUSSION TITLE
Maximum & minimum supported number overflow

### DISCUSSION THREAD

**Discussion by dakota-kallas** [uNmAnNeR/imaskjs#894] [Q&A] [answered]
**Describe the bug**
This bug is related to the core functionality of the plugin. JavaScript supports up to a maximum integer value of **2^53**. I want to use the numeric mask, but am running into cases where I am exceeding this limit. The plugin is casting my values down or up to the maximum or minimum safe integer supported by JS. This is preventing me from being able to use the numeric mask. It would be nice if this used something like a BigNumber type or was done using string manipulation of some kind so that it would support more than the maximum and minimum number values or a poly fill of a bigger type.

The plugin is expecting a `number` type for both `min` and `max` when using a numeric mask. This is a part of the problem.

**CodePen**
https://codepen.io/dakota-kallas/pen/vYQxeJb?editors=1010

**Environment:**
 - OS: Windows 10
 - Browser: Chrome v114.0.5735.134
 - IMask version: 7.0.1
 - Framework/plugin version: Vanilla plugin

**Comment by uNmAnNeR** [comment]
hi, i don't think it's a problem. Number mask works with number type. That's fine.
If you need more then you can try function mask or use combo regexp + validation. No big deal. IMask has a good start for any custom solution.

**Reply by dakota-kallas** [reply]
Do you have any suggestions, @uNmAnNeR, as to how I would go about creating a mask that works identical to a number mask, just supporting a number larger than 2^53?

**Reply by uNmAnNeR** [reply]
Okay, i have a suggestions.
Start with something like https://codepen.io/uNmAnNeR/pen/yLQbvYy?editors=1010
may be it already solves your problem.
if not: overwrite `format`/`parse` callbacks to handle string <-> YOUR-TYPE conversion
if still not: inspect sources for number mask, remove validation that you don't need and create your custom mask

**Reply by dakota-kallas** [reply, marked as answer]
Not quite the solution I was looking for, I ended up creating a fork to change how the numeric mask works. It changes the max and min values to be treated like string that allow you to set a maximum and minimum about of digits allowed before/after the decimal point (https://github.com/dakota-kallas/imaskjs). I couldn't find a way to use my forked repo as a dependency ( #892) properly with npm, so I had to change around the file structure a bit as well.

**Reply by uNmAnNeR** [reply]
@dakota-kallas ultimatively :)
